scripts/CMSRunAnalysis.py

```diff
@@ -698,15 +698,42 @@ def StripReport(report):
         rep = Report("cmsRun")
         rep.parse('FrameworkJobReport.xml', "cmsRun")
         jobExitCode = rep.getExitCode()
+        print("Job Exit Code from FrameworkJobReport.xml: %s " % jobExitCode)
         rep = rep.__to_json__(None)
         with open('WMArchiveReport.json', 'w') as of:
             json.dump(rep, of)
         StripReport(rep)
         # Record the payload process's exit code separately; that way, we can distinguish
         # cmsRun failures from stageout failures.  The initial use case of this is to
         # allow us to use a different LFN on job failure.
+        if jobExitCode == 0 and applicationExitCode > 0:
+            # We need to consider applicationExitCode as well, not only the exitcode from FWJR.xml
+            # ref: https://github.com/dmwm/CRABServer/issues/7571
+            # A brief recap:
+            # - the FWJR.xml does not contain an exit code if the application completed successfully
+            # - getExitCode() will default to 0 if it can not find an exit code
+            #   in a syntactically correct valid FWJR.
+            # - For CMSSW versions earlier than 12_6_X, it is possible that the
```

Comment:
change: earlier than 12_6_x --> 12_6_x or higher

Reply:
I improved the comments. thanks!